### ISSUE TITLE
Inconsistent JSON log timestamp formats

### ISSUE THREAD

**Issue by peterpramb** [elastic/elasticsearch#44816]
**Elasticsearch version** (`bin/elasticsearch --version`):
`Version: 7.2.0, Build: default/tar/508c38a/2019-06-20T15:54:18.811730Z, JVM: 11.0.3`

**Plugins installed**:
`n/a`

**JVM version** (`java -version`):
`OpenJDK version "11.0.3" 2019-04-16`

**OS version** (`uname -a` if on a Unix-like system):
`Linux 3.10.0-957.21.3.el7.x86_64`

**Description of the problem including expected versus actual behavior**:
All JSON log files use the same timestamp format (ISO8601+TZ), except for the audit log, which lacks TZ information. This is due to the format spec `%d{ISO8601}` in log4j2.properties, which makes parsing with e.g. Filebeat difficult.

Examples:
- `"timestamp": "2019-07-22T12:55:29,580"` (audit.json)
- `"timestamp": "2019-07-24T15:43:26,008+0200"` (server.json)

According to the log4j2 documentation the format should be `ISO8601_OFFSET_DATE_TIME_HHMM` to include the TZ information, but this does not seem to work as intended (produces the default format `2019-07-24 15:38:04,574`, so the included log4j2 version is probably too old).

I had to explicitly specify the format to get complete timestamps:
```
appender.audit_rolling.layout.pattern = {\
                "@timestamp":"%d{yyyy-MM-dd'T'HH:mm:ss,SSSXX}"\
...
```

Please consider unifying the JSON log timestamps.

**Comment by pgomulka** [comment]
thank you for rising this @peterpramb 
This was fixed in https://github.com/elastic/elasticsearch/issues/42666 PR https://github.com/elastic/elasticsearch/pull/42671
you can use this fix once you upgrade to 7.3

**Comment by peterpramb** [comment]
Thanks for the pointer - completely missed `@timestamp` vs. `timestamp`...

Successfully "backported" to our 7.2.0.
